File: DATA_STRUCTURES/stack_quee/min_stack.py
class MinStack(object):

    def __init__(self):
        self.min = []
        self.array = []
        self.size = 0
    def push(self, val):
        """
        :type val: int
        :rtype: None
        """
        self.array.append(val)
        

    def pop(self):
        """
        :rtype: None
        """
        self.array.pop()
    def top(self):
        """
        :rtype: int
        """
        return self.array[len() -1]
        

    def getMin(self):
        """
        :rtype: int
        """
        return min(self.array)
        

# Your MinStack object will be instantiated and called as such:
val = 90
obj = MinStack()
obj.push(val)
obj.push(val)
obj.push(val)
param_4 = obj.getMin()
obj.pop()
param_3 = obj.top()
param_4 = obj.getMin()

File: DATA_STRUCTURES/stack_quee/min_stack.java
package stack_quee;
import java.lang.Math;
class MinStack {
    int arr [];
    int size =0;
    MinStack min = new MinStack();
    public MinStack() {
        min.push((int)Math.pow(2,32));
        this.arr = new int [9];
    }
    
    public void push(int val) {
        this.arr[size ]  = val;
        if(val <this.min.top()){
            this.min.push(val);
        }
        this.size++;
    }
    
    public void pop() {
        this.size --;
        if (this.min.top() == this.top()){
            this.min.pop();
        }
        this.pop();
    }
    
    public int top() {
        if (this.arr != null){
            return this.arr[this.size-1];
        }
        return -1;
    }
    
    public int getMin() {
        return this.min.top();
    }
}

/**
 * Your MinStack object will be instantiated and called as such:*/
public class min_stack{
    public static void main(String[] args) {
        int val = 90 ; 
        MinStack obj = new MinStack();
        obj.push(val);
        obj.pop();
        int param_3 = obj.top();
        int param_4 = obj.getMin();
         
    }
}
